Source/Dafny/Resolver.cs

```diff
@@ -13546,10 +13546,16 @@ private class RBranchExpr : RBranch {
           callee.IsStatic ? null : s.Receiver);
         // type check the out-parameter arguments (in-parameters were type checked as part of ResolveActualParameters)
         for (int i = 0; i < callee.Outs.Count && i < s.Lhs.Count; i++) {
-          var it = callee.Outs[i].Type;
+          var outFormal = callee.Outs[i];
+          var it = outFormal.Type;
           Type st = SubstType(it, typeMap);
           var lhs = s.Lhs[i];
-          AddAssignableConstraint(s.Tok, lhs.Type, st, "incorrect type of method out-parameter" + (callee.Outs.Count == 1 ? "" : " " + i) + " (expected {1}, got {0})");
+          AddAssignableConstraint(
+            s.Tok, lhs.Type, st, "incorrect type of method out-parameter" +
```

Comment:
I think this expression could use some variable extraction, and an interpolated string.

```
$"incorrect type of method out-parameter{parameterReference} (expected {{1}}, got {{0}})"
```

Reply:
Nice use of double braces. Will do it.